src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,132 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.exceptions.InvalidFolderException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * <code>Storage</code> class handles the loading of
+ * tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+
+    /**
+     * Constructor for Storage class.
+     * Initializes File for the task file.
+     *
+     * @param filePath Filepath of the file that saves task
+     * @throws InvalidFolderException If folder to store the file does not exist.
+     */
+    public Storage(String filePath) throws InvalidFolderException {
+        try {
+            File file = new File(filePath);
+
+            if (!file.exists()) {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+            } else {
+                readFileContents(filePath);
+            }
+        } catch (IOException ex) {
+            throw new InvalidFolderException();
+        }
+    }
+
+    /**
+     * Returns a List of existing tasks from the file when Duke starts up.
+     *
+     * @param filePath Filepath of the file that saves task.
+     * @return List of existing tasks in the file.
+     * @throws FileNotFoundException If the file cannot be found.
+     */
+    public List<Task> readFileContents(String filePath)
+            throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        List<Task> list = new ArrayList<>();
```

Comment:
Consider renaming this variable to further specify what the contents of the list are?